### PR TITLE
feat: add ProductSalesDetailsScreen

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#80344

### CODE REVIEW

File: packages/apps/sales/src/components/molecules/DropdownProductSale/DropdownProductSale.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import {formatDate, useSelector, useTranslator} from '@axelor/aos-mobile-core';
+import {
+  Badge,
+  LabelText,
+  checkNullString,
+  usePriceFormat,
+  useThemeColor,
+} from '@axelor/aos-mobile-ui';
+
+const DropdownProductSale = ({isProductCompanyConfig}) => {
```
Comment (vhu-axelor) [inline packages/apps/sales/src/components/molecules/DropdownProductSale/DropdownProductSale.tsx:30]: Type it

File: packages/apps/sales/src/components/molecules/DropdownProductTypology/DropdownProductTypology.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useCallback} from 'react';
+import {View} from 'react-native';
+import {useSelector, useTranslator, useTypes} from '@axelor/aos-mobile-core';
+import {LabelText, checkNullString} from '@axelor/aos-mobile-ui';
+
+const DropdownProductTypology = ({isProductCompanyConfig}) => {
```
Comment (vhu-axelor) [inline packages/apps/sales/src/components/molecules/DropdownProductTypology/DropdownProductTypology.tsx:24]: Same here

File: packages/apps/sales/src/components/molecules/ProductHeader/ProductHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {StyleSheet, View} from 'react-native';
+import {useMetafileUri, useSelector} from '@axelor/aos-mobile-core';
+import {Image, Text, TextUnit, usePriceFormat} from '@axelor/aos-mobile-ui';
+import {TaxModeBadge} from '../../atoms';
+
+const ProductHeader = ({isProductCompanyConfig}) => {
```
Comment (vhu-axelor) [inline packages/apps/sales/src/components/molecules/ProductHeader/ProductHeader.tsx:25]: Same here

File: packages/apps/sales/src/components/organisms/ProductDropdownCard/ProductDropdownCard.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import {StyleSheet, View} from 'react-native';
+import {DropdownCardSwitch} from '@axelor/aos-mobile-ui';
+import {useSelector, useTranslator} from '@axelor/aos-mobile-core';
+import {
+  DropdownMultipleQuantities,
+  DropdownProductSale,
+  DropdownProductTypology,
+} from '../../molecules';
+
+const ProductDropdownCard = ({isProductCompanyConfig}) => {
```
Comment (vhu-axelor) [inline packages/apps/sales/src/components/organisms/ProductDropdownCard/ProductDropdownCard.tsx:29]: Same here

File: packages/apps/sales/src/components/organisms/ProductDropdownCard/ProductDropdownCard.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    </View>
+  );
+};
+
+export default ProductDropdownCard;
+
+const styles = StyleSheet.create({
+  container: {
+    alignItems: 'center',
+  },
```
Comment (vhu-axelor) [inline packages/apps/sales/src/components/organisms/ProductDropdownCard/ProductDropdownCard.tsx:79]: Not useful

File: packages/apps/sales/src/components/templates/CompanySearchBar/CompanySearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {displayItemName, useSelector} from '@axelor/aos-mobile-core';
+import {AutoCompleteSearch} from '@axelor/aos-mobile-ui';
+
+const CompanySearchBar = ({style, setCompany, company}) => {
```
Comment (vhu-axelor) [inline packages/apps/sales/src/components/templates/CompanySearchBar/CompanySearchBar.tsx:23]: Type it

File: packages/apps/sales/src/i18n/en.json [hunk 7]
```diff
@@ -56,11 +58,36 @@
...
+  "Sales_SeePriceLists": "See price lists",
+  "Sales_InternalDescription": "Internal description",
+  "Sales_ProductTypology": "Product typology",
+  "Sales_MultipleQuantities": "Multiple quantities",
+  "Sales_Type": "Type",
+  "Sales_SubType": "Sub type",
+  "Sales_ProductFamily": "Product family",
+  "Sales_CategoryFamily": "Category family",
+  "Sales_ProcurementMethod": "Procurement method",
+  "Sales_PriceWT": "Prix de vente HT",
```
Comment (vhu-axelor) [inline packages/apps/sales/src/i18n/en.json:72]: "Sale price W.T."

File: packages/apps/sales/src/i18n/fr.json [hunk 8]
```diff
@@ -56,11 +58,36 @@
...
+  "Sales_SeePriceLists": "Voir les listes de prix",
+  "Sales_InternalDescription": "Description interne",
+  "Sales_ProductTypology": "Typologie produit",
+  "Sales_MultipleQuantities": "Quantités multiples",
+  "Sales_Type": "Type",
+  "Sales_SubType": "Sous-type",
+  "Sales_ProductFamily": "Famille de produits",
+  "Sales_CategoryFamily": "Famille de catégories",
+  "Sales_ProcurementMethod": "Méthode d'approvisionnement",
+  "Sales_PriceWT": "Sale price WT",
```
Comment (vhu-axelor) [inline packages/apps/sales/src/i18n/fr.json:72]: "Prix de vente H.T."

File: packages/apps/sales/src/i18n/en.json [hunk 9]
```diff
@@ -56,11 +58,36 @@
...
+  "Sales_SaleUnit": "Sale unit",
+  "Sales_LaunchDate": "Launch date",
+  "Sales_Unrenewed": "Unrenewed",
+  "Sales_Prototype": "Prototype",
+  "Sales_finishedProduct": "Finished product",
+  "Sales_SemiFinishedProduct": "Semi-finished product",
+  "Sales_Component": "Component",
+  "Sales_NoConfigPerCompany": "No product configuration per company",
+  "Sales_QuantitiesAreRestricted": "Les quantités sont restreintes",
+  "Sales_ProductPulledOfMarket": "Product pulled off market Date",
```
Comment (vhu-axelor) [inline packages/apps/sales/src/i18n/en.json:83]: "Product pulled off market date"

File: packages/apps/sales/src/i18n/fr.json [hunk 10]
```diff
@@ -56,11 +58,36 @@
...
+  "Sales_SaleCurrency": "Devise de vente",
+  "Sales_SaleUnit": "Unité de vente",
+  "Sales_LaunchDate": "Date de mise en application",
+  "Sales_Unrenewed": "Non renouvelé",
+  "Sales_Prototype": "Prototype",
+  "Sales_finishedProduct": "Produit fini",
+  "Sales_SemiFinishedProduct": "Produit semi-finit",
```
Comment (vhu-axelor) [inline packages/apps/sales/src/i18n/fr.json:79]: "Produit semi-fini"

File: packages/apps/sales/src/screens/product/ProductSalesDetailsScreen.tsx [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+        />
+      )}
+      <CompanySearchBar
+        company={company}
+        setCompany={setCompany}
+        style={isProductCompanyConfig && styles.searchBar}
+      />
+      <ProductDescription />
+      <ProductDropdownCard isProductCompanyConfig={isProductCompanyConfig} />
+      <ProductFloatingButton />
```
[the remark this answers is not in the document]
Comment (gca-axelor) [reply packages/apps/sales/src/screens/product/ProductSalesDetailsScreen.tsx]: I created a ticket to fix FloatingButton component

File: packages/apps/sales/src/components/molecules/DropdownProductTypology/DropdownProductTypology.tsx [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+  const {Product} = useTypes();
+
+  const {product, productCompany} = useSelector(
+    (state: any) => state.sales_product,
+  );
+
+  const renderLabelText = useCallback(
+    (titleKey: string, value: string | number) => {
+      if (!checkNullString(value)) {
+        return <LabelText title={`${I18n.t(titleKey)} :`} value={value} />;
```
[the remark this answers is not in the document]
Comment (gca-axelor) [reply packages/apps/sales/src/components/molecules/DropdownProductTypology/DropdownProductTypology.tsx]: i will create a ticket for this

**Pull request merged**